app/src/main/java/org/greenstand/android/TreeTracker/messages/ChatScreen.kt

```diff
@@ -0,0 +1,286 @@
+                        autoCorrect = false,
+                    ),
+                    placeholder = {
+                        Text(
+                            text = stringResource(id = R.string.click_to_write_message),
+                            color = Color.White
+                        )
+                    },
+                    keyboardActions = KeyboardActions(
+                    onGo = {
+                        viewModel.sendMessage()
+                    }
+                ),
+                colors = TextFieldDefaults.textFieldColors(
+                    textColor = AppColors.LightGray,
+                    backgroundColor = AppColors.DeepGray,
+                    focusedIndicatorColor = Color.Transparent,
+                    unfocusedIndicatorColor = Color.Transparent,
+                    disabledIndicatorColor = Color.Transparent
+                ),
+                    maxLines = 2
+                )
+            }
+    }
+}
+}
+
+const val ConversationTestTag = "ConversationTestTag"
+
+@Composable
+fun Messages(
+    messages: Flow<List<DirectMessage>>?,
+    scrollState: LazyListState,
+    modifier: Modifier = Modifier
+) {
+    val scope = rememberCoroutineScope()
+    Box(modifier = modifier) {
+        val authorAdmin = "admin"
+        LazyColumn(
+            reverseLayout = true,
+            state = scrollState,
+            contentPadding = PaddingValues(top = 10.dp),
+            // Add content padding so that the content can be scrolled (y-axis)
+            // below the status bar + app bar
+            // TODO: Get height from somewhere
+            modifier = Modifier
+                .testTag(ConversationTestTag)
+                .fillMaxSize()
+        ) {
+            messages?.map { message ->
+                for (index in message.indices) {
+                    val prevAuthor = message.getOrNull(index - 1)?.from
+                    val nextAuthor = message.getOrNull(index + 1)?.from
+                    val content = message[index]
+                    val isFirstMessageByAuthor = prevAuthor != content.from
+                    val isLastMessageByAuthor = nextAuthor != content.from
+                    item {
+                        Message(
+                            msg = content,
+                            isAdmin = content.from == authorAdmin,
+                            isFirstMessageByAuthor = isFirstMessageByAuthor,
+                            isLastMessageByAuthor = isLastMessageByAuthor
+                        )
+                    }
+                }
+
+            }
+        }
+        }
+}
+
+@Composable
+fun Message(
+    msg: DirectMessage,
+    isAdmin: Boolean,
+    isFirstMessageByAuthor: Boolean,
+    isLastMessageByAuthor: Boolean
+) {
+    val spaceBetweenAuthors = if (isLastMessageByAuthor) Modifier.padding(top = 8.dp) else Modifier
+    Row(modifier = spaceBetweenAuthors) {
+        TextMessage(
+            msg = msg,
+            isAdmin = isAdmin,
```

Comment:
I'd avoid using "admin" in the code. It could be possible later that it's not an admin

Reply:
yes, don't think of the the incoming messages as from 'admin' but rather from another user.  for the time being it's a manager that will send messages, but we may allow direct person-to-person messaging at some point.